macaroons/store.go

```diff
@@ -157,17 +164,27 @@ func (r *RootKeyStorage) Get(_ context.Context, id []byte) ([]byte, error) {
 
 // RootKey implements the RootKey method for the bakery.RootKeyStorage
 // interface.
-// TODO(aakselrod): Add support for key rotation.
-func (r *RootKeyStorage) RootKey(_ context.Context) ([]byte, []byte, error) {
+func (r *RootKeyStorage) RootKey(ctx context.Context) ([]byte, []byte, error) {
 	r.encKeyMtx.RLock()
 	defer r.encKeyMtx.RUnlock()
 
 	if r.encKey == nil {
 		return nil, nil, ErrStoreLocked
 	}
 	var rootKey []byte
-	id := defaultRootKeyID
-	err := kvdb.Update(r, func(tx kvdb.RwTx) error {
+
+	// Read the root key ID from the context. If no key is specified in the
+	// context, an error will be returned.
+	id, err := RootKeyIDFromContext(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if bytes.Equal(id, encryptedKeyID) {
```

Comment:
We should also check for an empty ID. Anyone using the code directly as a library could still pass in a context with an empty value.

Reply:
an empty key won't pass as the following `ns.Put(id, encKey)` will return an error. But sure we can return early and be explicit. The check is added in `RootKeyIDFromContext `, and a test against empty value is added.